### ISSUE TITLE
Player's position is off on each connection to server.

### ISSUE THREAD

**Issue by Zombs-R-Cute** [SmartlyDressedGames/Unturned-3.x-Community#4813]
When logging back in to a server the position where you logged out is off.  If you repeat this, without moving, the position your standing will continually change.
![Screenshot_20241214_094144](https://github.com/user-attachments/assets/333b813c-70bb-4e1a-a7a6-149dd64ecff8)
![Screenshot_20241214_094220](https://github.com/user-attachments/assets/6b8dc9ab-f8f5-430f-b6c2-eaaa0c5fa366)
![Screenshot_20241214_094251](https://github.com/user-attachments/assets/3093f56e-fc98-4523-b00d-6db69b08938f)
![Screenshot_20241214_094426](https://github.com/user-attachments/assets/bca3237b-80dd-4166-be77-685e1e86266f)

**Comment by SDGNelson** [comment]
That's weird, thanks for the report! Taking a quick peek at `Player.savePositionAndRotation` and `Provider.loadPlayerSpawn` the only offset that jumps out at me is 1 cm upward. A plugin *could* be offsetting using `Provider.onLoginSpawning`, but that seems unlikely. I'll try reproducing this but probably not for this coming week's update.

**Comment by SDGNelson** [comment]
Funnily enough it turns out the reason for moving sideways was... input defaulted to walking sideways! (Defaulted to zero — zero is left, one is neutral, and two is right.)

This uncovered a separate issue in saving the player position. If you were in a vehicle it *should've* been using the exit position, but instead it was using your character position. Likely causing some unintentional teleports to spawnpoint when loaded position was blocked by the vehicle.
